src/main/webapp/app/exercises/programming/shared/instructions-render/programming-exercise-instruction.component.ts

```diff
@@ -312,6 +309,39 @@ export class ProgrammingExerciseInstructionComponent implements OnChanges, OnDes
         }
     }
 
+    private renderMarkdown(): void {
+        // Highlight differences between previous and current markdown
+        if (
+            this.examExerciseUpdateHighlighterComponent?.showHighlightedDifferences &&
+            this.examExerciseUpdateHighlighterComponent.outdatedProblemStatement &&
+            this.examExerciseUpdateHighlighterComponent.updatedProblemStatement
+        ) {
+            let outdatedMarkdown = htmlForMarkdown(this.examExerciseUpdateHighlighterComponent.outdatedProblemStatement, this.markdownExtensions);
+            outdatedMarkdown = this.programmingExerciseTaskWrapper.injectTasksIntoHTML(outdatedMarkdown);
+            let updatedMarkdown = htmlForMarkdown(this.examExerciseUpdateHighlighterComponent.updatedProblemStatement, this.markdownExtensions);
+            updatedMarkdown = this.programmingExerciseTaskWrapper.injectTasksIntoHTML(updatedMarkdown);
+            const diffedMarkdown = diff(outdatedMarkdown, updatedMarkdown);
+            this.renderedMarkdown = this.sanitizer.bypassSecurityTrustHtml(diffedMarkdown);
+            // Differences between UMLs are ignored, and we only inject the current one
+            setTimeout(() => {
+                const injectUML = this.injectableContentForMarkdownCallbacks[this.injectableContentForMarkdownCallbacks.length - 1];
+                if (injectUML) {
+                    injectUML();
+                }
+            }, 0);
+        } else {
+            this.injectableContentForMarkdownCallbacks = [];
+            this.renderedMarkdown = this.markdownService.safeHtmlForMarkdown(this.exercise.problemStatement, this.markdownExtensions);
```

Comment:
I had an error in my console here. It didn't cause any problems, but can you double-check if `this.exercise` is ever `undefined` when this method is called?

It seems to happen when I first open the problem statement of a programming exercise.

![image](https://github.com/ls1intum/Artemis/assets/38403547/caf2c7d4-c171-4c24-af74-c1c95a769c85)

Reply:
Regarding URLs, I don't think there's something we can do about it. I can open an issue on the library's github and maybe they'll find a way to handle this case in next versions.

When it comes to the feedback, I managed to make it clickable again, both for new version and version with diffs.

I also added check for undefined, the error shouldn't appear in the console anymore.